impl/src/main/java/org/jboss/shrinkwrap/impl/BeansXml.java

```diff
@@ -239,9 +251,19 @@ public BeansXml setBeanDiscoveryMode(BeanDiscoveryMode mode) {
     @Override
     public InputStream openStream() {
         StringBuilder xml = new StringBuilder();
-        xml.append("<beans version=\"" + getBeansXmlVersion().getValue() + "\" bean-discovery-mode=\"");
-        xml.append(getBeanDiscoveryMode().getValue());
-        xml.append("\">\n");
+        xml.append("<beans");
+        BeansXmlVersion version = getBeansXmlVersion();
+        // add version if required
+        if (version != null) {
+            xml.append(" version=\"" + getBeansXmlVersion().getValue() + "\"");
```

Comment:
Redundant `getBeansXmlVersion()` call:

```suggestion
            xml.append(" version=\"" + version.getValue() + "\"");
```